app/controllers/KairosDBProxy.java

```diff
@@ -0,0 +1,349 @@
+        return proxy();
+    }
+
+    /**
+     * Proxied version call.
+     *
+     * @return Proxied version response.
+     */
+    public CompletionStage<Result> version() {
+        return proxy();
+    }
+
+    /**
+     * Caching metricNames call.
+     *
+     * @param containing simple string match filter for metric names
+     * @return Cached metric names, filtered by the query string.
+     */
+    public CompletionStage<Result> metricNames(final String containing) {
+        final List<String> metrics = _cache.getIfPresent(METRICS_KEY);
+
+        final CompletionStage<List<String>> response;
+        if (metrics != null) {
+            response = CompletableFuture.completedFuture(metrics);
+        } else {
+            // Refresh the cache
+            final CompletionStage<List<String>> queryResponse = _kairosDbClient.queryMetricNames()
+                    .thenApply(KairosMetricNamesQueryResponse::getResults)
+                    .thenApply(list -> {
+                        _cache.put(METRICS_KEY, list);
+                        _metricsList.set(list);
+                        return list;
+                    });
+
+            final List<String> metricsList = _metricsList.get();
+            if (metricsList != null) {
+                response = CompletableFuture.completedFuture(metricsList);
+            } else {
+                response = queryResponse;
+            }
+        }
+
+        return response
+                .thenApply(list -> filterMetricNames(list, containing))
+                .thenApply(list -> new KairosMetricNamesQueryResponse.Builder().setResults(list).build())
+                .<JsonNode>thenApply(_mapper::valueToTree)
+                .thenApply(Results::ok);
+    }
+
+    private static List<String> filterMetricNames(final List<String> metricNames, final String containing) {
+        if (containing == null || containing.isEmpty()) {
+            return metricNames;
+        }
+
+        final String lowerContaining = containing.toLowerCase(Locale.ENGLISH);
+        final Predicate<String> containsPredicate = s -> s.toLowerCase().contains(lowerContaining);
+        final Predicate<String> isPT1M = s -> s.startsWith("PT1M/");
+
+        return metricNames.stream()
+                .filter(isPT1M.negate())
+                .filter(containsPredicate)
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Proxy a request.
+     *
+     * @return the proxied {@link Result}
+     */
+    private CompletionStage<Result> proxy() {
+        final String path = request().uri();
+        LOGGER.info().setMessage("proxying call from " + path + " to kairosdb").log();
+        final CompletableFuture<Result> promise = new CompletableFuture<>();
+        final Http.Request request = request();
+        final boolean isHttp10 = request.version().equals("HTTP/1.0");
+        LOGGER.info().setMessage(String.format("Version=%s", request.version())).log();
```

Comment:
You should put data into data fields instead of the message

Reply:
Removed this line.